### ISSUE TITLE
1.6.7 release mock endpoint broken on ruby 1.9.3

### ISSUE THREAD

**Issue by dereklwood** [rack/rack#1168]
Hi,
[This commit](https://github.com/rack/rack/commit/c7ee52f6ba4684ce917b97c1ff4c7c523760a9d2), which appears to have been released in version 1.6.7, invokes the ``String#b`` method which is not present in Ruby 1.9.3 (https://ruby-doc.org/core-1.9.3/String.html).

It looks like the code attempts to protect against this exact failure by [checking](https://github.com/rack/rack/commit/c7ee52f6ba4684ce917b97c1ff4c7c523760a9d2#diff-4fc9bc1f7d91630f4f9f47fc6663f3f7R146) for the ``:encoding`` method, but Ruby 1.9.3 does indeed [respond to this method](https://ruby-doc.org/core-1.9.3/String.html#method-i-encoding), and so it proceeds to fail on the ``.b`` invocation.  (Ruby 1.8 has neither the ``:encoding`` or ``:b`` methods.)

```
Failures:

  1) Sinatra API should serve status endpoint
     Failure/Error: get '/status'
     
     NoMethodError:
       undefined method `b' for "GET":String
     # /Users/derek/.rvm/gems/ruby-1.9.3-p374/gems/rack-1.6.7/lib/rack/mock.rb:148:in `env_with_encoding'
     # /Users/derek/.rvm/gems/ruby-1.9.3-p374/gems/rack-1.6.7/lib/rack/mock.rb:94:in `env_for'
     # /Users/derek/.rvm/gems/ruby-1.9.3-p374/gems/rack-test-0.6.3/lib/rack/test.rb:237:in `env_for'
     # /Users/derek/.rvm/gems/ruby-1.9.3-p374/gems/rack-test-0.6.3/lib/rack/test.rb:57:in `get'
     # ./spec/api_spec.rb:31:in `block (2 levels) in <top (required)>'
```

**Comment by segiddins** [comment]
The failures on CI are visible at https://travis-ci.org/rack/rack/jobs/231717923

**Comment by rafaelfranca** [comment]
cc @eileencodes

**Comment by eileencodes** [comment]
Thanks @rafaelfranca, taking a look 👀 

cc/ @tenderlove

**Comment by eileencodes** [comment]
This (and an issue on 1.8.7 that we missed) should be fixed with the 1.6.8 release. Thanks for the report @dereklwood and @segiddins!

**Comment by dereklwood** [comment]
Thanks for the quick turnaround!
